core/src/main/java/org/apache/brooklyn/location/localhost/LocalhostMachineProvisioningLocation.java

```diff
@@ -268,22 +266,15 @@ public void release(SshMachineLocation machine) {
                 BrooklynConfigKeys.SKIP_ON_BOX_BASE_DIR_RESOLUTION, 
                 true);
 
-        private static final WithMutexes mutexSupport = new MutexSupport();
```

Comment:
You've change the semantics for localhost - note that this field was static. To preserve that, you can override `WithMutexes mutexes()` to return the static field.

I think you'll be fine removing from the constructor the bit that sets it in the map (note that `SshMachineLocation.mutexSupport` was marked with `@SetFromFlag`). By just overriding `mutexes()`, I think it should be enough.

Reply:
Excellent catch, thanks.